src/transpile/fusion.hpp

```diff
@@ -658,7 +658,7 @@ bool DiagonalFusion::aggregate_operations(oplist_t& ops,
       continue;
 
     std::vector<uint_t> fusing_op_idxs;
-    for (; op_idx < next_diagonal_start; ++op_idx)
+    for (; op_idx < next_diagonal_start && op_idx < fusion_end; ++op_idx)
```

Comment:
This looks like odd syntax for a `for` loop. Should it be changed to a `while` loop instead?